src/lib/__tests__/generateChartImg.test.js

```diff
@@ -0,0 +1,104 @@
+const generateChartImg = require('../generateChartImg');
+
+describe('generateChartImg function', () => {
+  test('returns a URL representing the chart image', async () => {
+    // Define input parameters
+    const type = 'line';
+    const data = [{ x: 1, y: 2 }, { x: 2, y: 3 }, { x: 3, y: 4 }];
+    const xLabel = 'X Axis';
+    const yLabel = 'Y Axis';
+    const title = 'Chart Title';
+    const color = 'blue';
+
+    // Call the generateChartImg function
+    const imgUrl = await generateChartImg(type, data, xLabel, yLabel, title, color);
+
+    // Check if the function returns a URL representing the chart image
+    expect(imgUrl).toMatch(/^blob:/);
+  });
+});
+
+
+
+describe('generateChartImg function', () => {
+    test('still makes request even if color is invalid', async () => {
+      // Define input parameters with an invalid color
+      const type = 'line';
+      const data = [{ x: 1, y: 2 }, { x: 2, y: 3 }, { x: 3, y: 4 }];
+      const xLabel = 'X Axis';
+      const yLabel = 'Y Axis';
+      const title = 'Chart Title';
+      const color = 29; // Invalid color
+
+      const imgUrl = await generateChartImg(type, data, xLabel, yLabel, title, color);
+
+  
+      
+      expect(imgUrl).toMatch(/^blob:/);
+    });
+  });
+
+
+  describe('generateChartImg function', () => {
+    test('returns a valid URL representing the chart image', async () => {
+      // Define input parameters
+      const type = 'line';
+      const data = [{ x: 1, y: 2 }, { x: 2, y: 3 }, { x: 3, y: 4 }];
+      const xLabel = 'X Axis';
+      const yLabel = 'Y Axis';
+      const title = 'Chart Title';
+      const color = 'blue';
+  
+      // Call the generateChartImg function
+      const imgUrl = await generateChartImg(type, data, xLabel, yLabel, title, color);
+  
+      
+      const parsedUrl = new URL(imgUrl);
+  
+      // Assert that a proper URL is returned
+      expect(parsedUrl).toBeDefined();
+    });
+  });
+
+
+
+  describe('generateChartImg function', () => {
+    test('returns a valid URL representing the chart image when color is not inputted', async () => {
+      // Define input parameters
+      const type = 'line';
+      const data = [{ x: 1, y: 2 }, { x: 2, y: 3 }, { x: 3, y: 4 }];
```

Comment:
Similar to the first comment, maybe for data have more points just to test that this works with large amounts of data.